utils/query-state/Select-UpstreamBranchGraph.psm1

```diff
@@ -0,0 +1,91 @@
+Import-Module -Scope Local "$PSScriptRoot/Get-UpstreamBranch.psm1"
+Import-Module -Scope Local "$PSScriptRoot/Get-GitFile.psm1"
+Import-Module -Scope Local "$PSScriptRoot/Configuration.psm1"
+
+function Get-GraphLayers($root) {
+	$uniqueNamesWithoutMain = $root `
+		| ForEach-Object { $_.name } `
+		| Where-Object { $_ -ne 'main' } `
+		| Sort-Object -Unique;
+
+	$output = ,$uniqueNamesWithoutMain
+	$elems = $root | ForEach-Object {
+		$_.connections | Where-Object { $_.name -ne 'main' };
+	}
+	if ($elems.Length -gt 0) {
+		$output += Get-GraphLayers $elems;
+	}
+	return $output;
+}
+
+function Write-GraphLayers($layers, [switch]$checkList) {
+	[array]::Reverse($layers);
+	$seen = New-Object 'System.Collections.Generic.HashSet[string]';
+
+	$layers | ForEach-Object {
+		$unseen = $_ | Where-Object { -not $seen.Contains($_) }
+		$to_write = $unseen -join ', ';
+		if ($to_write.Length -gt 180 -and -not $checkList) {
+			Write-Host ($unseen -join "`r`n")
+		} elseif ($checkList) {
+			$unseen | ForEach-Object {
+				Write-Host "[ ] $_"
+			}
+		} else {
+			Write-Host $to_write
+		}
+		$unseen | ForEach-Object {
+			$seen.Add($_) > $null;
+		}
+		Write-Host ("=" * 24)
+		Write-Host
+	}
+}
+
+function Select-UpstreamBranchGraph([String]$branchName, [switch]$checkList=$false) {
+	$config = Get-Configuration;
+	if ($branchName -eq $config.defaultServiceLine) {
+		Write-Host "Cannot graph $($config.defaultServiceLine), it is the default service line and should not have upstreams";
```

Comment:
FYI, service lines can have upstreams, just don't in anything simpler than the "Multiple maintained versions" flow. That is, `line/2.1` would have an upstream of `line/2.0` which would have an upstream of `line/1.4`, etc.

The default service line should just be the user's default; that's why it's not stored in a global config. (I could see a well-informed team using a "service line" for a big future release if it has thorough internal QA before merging into it.)